rfqmath/fixed_point_test.go

```diff
@@ -159,6 +159,144 @@ func testFromUint64[N Int[N]](t *rapid.T) {
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](85_000, 3),
+			secondFp:     FixedPointFromUint64[N](100_000, 2),
+			tolerancePpm: 176470, // 17.65% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where secondFp is 15% less than firstFp,
+			// tolerance allows 10% (100000 PPM). Diff outside
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](85_000, 3),
+			secondFp:     FixedPointFromUint64[N](100_000, 2),
+			tolerancePpm: 100000, // 10% tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where firstFp and secondFp are equal,
+			// tolerance is 0 PPM. Diff within bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 2),
+			secondFp:     FixedPointFromUint64[N](100_000, 4),
+			tolerancePpm: 0, // 0% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where secondFp is 1% more than firstFp,
+			// tolerance allows 0.99% (9900 PPM). Diff outside
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 2),
+			secondFp:     FixedPointFromUint64[N](101_000, 3),
+			tolerancePpm: 9900, // 0.99% tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where secondFp is 5% less than firstFp,
+			// tolerance allows 5% (50000 PPM). Diff within bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 1),
+			secondFp:     FixedPointFromUint64[N](95000, 2),
+			tolerancePpm: 50000, // 5% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where secondFp is 10% less than firstFp,
+			// tolerance allows 9% (90000 PPM). Diff outside bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 3),
+			secondFp:     FixedPointFromUint64[N](90_000, 1),
+			tolerancePpm: 90000, // 9% tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where secondFp is 9% less than firstFp,
+			// tolerance allows 10% (100000 PPM). Diff within
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 4),
+			secondFp:     FixedPointFromUint64[N](91_000, 1),
+			tolerancePpm: 100000, // 10% tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where both prices are zero, should be within
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](0, 0),
+			secondFp:     FixedPointFromUint64[N](0, 0),
+			tolerancePpm: 100000, // any tolerance in PPM
+			withinBounds: true,
+		},
+		{
+			// Case where firstFp is zero and secondFp is
+			// non-zero, should not be within bounds.
+			firstFp:      FixedPointFromUint64[N](0, 0),
+			secondFp:     FixedPointFromUint64[N](100_000, 4),
+			tolerancePpm: 100000, // any tolerance in PPM
+			withinBounds: false,
+		},
+		{
+			// Case where secondFp is zero and firstFp is
+			// non-zero, should not be within bounds.
+			firstFp:      FixedPointFromUint64[N](100_000, 4),
+			secondFp:     FixedPointFromUint64[N](0, 0),
+			tolerancePpm: 100000, // any tolerance in PPM
+			withinBounds: false,
+		},
+	}
+
+	// Run the test cases.
+	for idx, tc := range testCases {
```

Comment:
would it make sense to add a case where there's a huge scale difference, in order to see precision getting exposed?

i.e
```
firstFp:      FixedPointFromUint64[N](314, 2),
secondFp:     FixedPointFromUint64[N](314_159_265_359, 11),
```